service/src/test/java/uk/gov/hmcts/reform/fpl/service/payment/PaymentServiceTest.java

```diff
@@ -99,6 +103,52 @@ void setup() {
             when(feeService.getFeesDataForC2(WITHOUT_NOTICE)).thenReturn(buildFeesData(feeForC2WithoutNotice));
         }
 
+        @Test
+        void shouldRetryPaymentsApiWhenInternalServerErrorThrown() {
+            CreditAccountPaymentRequest expectedPaymentRequest = testCreditAccountPaymentRequestBuilder()
+                .customerReference("1")
+                .amount(feeForC2WithNotice.getCalculatedAmount())
+                .fees(List.of(feeForC2WithNotice))
+                .build();
+
+            when(paymentApi.createCreditAccountPayment(AUTH_TOKEN, SERVICE_AUTH_TOKEN, expectedPaymentRequest))
+                .thenThrow(FeignException.InternalServerError.class);
+
+            assertThrows(Exception.class, () -> paymentService.callPaymentsApi(expectedPaymentRequest));
```

Comment:
I would check exact exception as this is part of method interface
assertThrows(PaymentException.class, () -> paymentService.callPaymentsApi(expectedPaymentRequest));
if method will throw different exception, then app would be broken and no email to cts will be sent